Be/snowman project edits (#53)
Ada-Developers-Academy/ada-precourse-v2: 01-intro-to-python/08-snowman.checkpoint.md

You will need to complete the `snowman(snowman_word)` function using the other functions that we built in previous exercises.  

> When you run the program it will give you the opportunity to run it against some automated tests, which check to see if, after several guesses you print out, "Congratuations, you win!" for success and "Sorry, you lose! The word was {snowman_word}" for failure.

Reviewer: (looks like this is correct in the replit)
```suggestion
When you run the program it will give you the opportunity to run it against some automated tests, which check to see if, after several guesses you print out, "Congratulations, you win!" for success and "Sorry, you lose! The word was {snowman_word}" for failure.
```

You should make use of the following functions from the previous lessons (already created for you):

- `print_snowman_graphic(num_wrong_guesses)` - This function prints out the appropriate snowman image depending on the number of wrong guesses the player has made.

> - `build_word_dict(snowman_word)` - This function takes snowman_word as input and returns a dictionary with a key-value pair for each letter in snowman_word where the key is the letter and the value is `False`.

Reviewer: Personally, I like a comma here, but it's probably not necessary.
```suggestion
- `build_word_dict(snowman_word)` - This function takes snowman_word as input and returns a dictionary with a key-value pair for each letter in snowman_word, where the key is the letter and the value is `False`.
```

> - `get_letter_from_user(snowman_word_dict, wrong_guesses_list)` - This function takes the snowman_word_dict and the list of characters that have been guessed incorrectly (wrong_guesses_list) as input. It asks for input from the user of a single character until a valid character is provided and then returns this character.

Reviewer: Possible confusion from input being used to describe both the parameters, as well as user input. I'd kind of like to refer to them as parameters, but at least I'd like to avoid using the same term to mean two different things in the same passage, and the other descriptions refer to the parameters as input. I also found the sentence describing what input is sought to be a little awkward.
```suggestion
- `get_letter_from_user(snowman_word_dict, wrong_guesses_list)` - This function takes the snowman_word_dict and the list of characters that have been guessed incorrectly (wrong_guesses_list) as input. It repeatedly asks the user to enter a single character until a valid character is provided. Once a valid character is entered, it is returned.
```

> - `get_word_progress(snowman_word, snowman_word_dict)` - This function takes the snowman_word and snowman_word_dict as input. It prints an output_string that shows the correct letter guess placements as well as the placements for the letters yet to be guessed. It returns True if all the letters of the word have been guessed, and False otherwise.

Reviewer: Not related to this review, but it irks me that this function prints as a side-effect. It sounds like a function whose main purpose is to get information about the progress, but also does output.

Author: Definitely something to consider for a larger refactor of the project

> - `get_word_progress(snowman_word, snowman_word_dict)` - This function takes the snowman_word and snowman_word_dict as input. It prints an output_string that shows the correct letter guess placements as well as the placements for the letters yet to be guessed. It returns True if all the letters of the word have been guessed, and False otherwise.

Reviewer: Is it possibly confusing to use the variable name `output_string` here? I know it's in the implementation, but that's not visible here. If we want the same description here AND in the replit, personally, I'd go without the underscore.
```suggestion
- `get_word_progress(snowman_word, snowman_word_dict)` - This function takes the snowman_word and snowman_word_dict as input. It prints an output string that shows the correct letter guess placements as well as the placements for the letters yet to be guessed. It returns True if all the letters of the word have been guessed, and False otherwise.
```
